audio_service/test/isolates_test.dart

```diff
@@ -4,241 +4,339 @@ import 'package:audio_service/audio_service.dart';
 
-    late bool spawningIsolate;
+  late bool spawningIsolate;
 
-    setUp(() async {
-      if (spawningIsolate) {
-        await runIsolateHandler();
-      } else {
-        IsolatedAudioHandler(_MockAudioHandler());
-      }
-      proxy = await IsolatedAudioHandler.lookup();
-    });
+  setUp(() async {
+    if (spawningIsolate) {
+      await runIsolateHandler();
+    } else {
+      IsolatedAudioHandler(_MockAudioHandler());
+    }
+    proxy = await IsolatedAudioHandler.lookup();
+  });
 
-    tearDown(() async {
-      await proxy.unregister();
-      if (spawningIsolate) {
-        await killIsolate();
-      }
+  tearDown(() async {
+    await proxy.unregister();
+    if (spawningIsolate) {
+      await killIsolate();
+    }
+  });
+
+  // We need to run the tests not only in different isolates but also in the
+  // same isolate since coverage is only collected by the main isolate:
+  // https://github.com/dart-lang/test/issues/1108
+  for (var differentIsolate in [true, false]) {
+    spawningIsolate = differentIsolate;
+    final isolateLabel =
+        differentIsolate ? '(different isolate)' : '(same isolate)';
+
+    test('$isolateLabel init', () async {
+      //expect(await proxy.playbackState.first, PlaybackState());
+      expect(await proxy.queue.first, const <MediaItem>[]);
+      expect(await proxy.queueTitle.first, '');
+      expect(await proxy.mediaItem.first, null);
+      expect(proxy.ratingStyle.hasValue, false);
+      expect(proxy.androidPlaybackInfo.hasValue, false);
+      expect(proxy.customState.hasValue, false);
     });
 
-    // We need to run the tests not only in different isolates but also in the
-    // same isolate since coverage is only collected by the main isolate:
-    // https://github.com/dart-lang/test/issues/1108
-    for (var differentIsolate in [true, false]) {
-      spawningIsolate = differentIsolate;
-      final isolateLabel =
-          differentIsolate ? ' (different isolate)' : ' (same isolate)';
-
-      test('init $isolateLabel', () async {
-        //expect(await proxy.playbackState.first, PlaybackState());
-        expect(await proxy.queue.first, const <MediaItem>[]);
-        expect(await proxy.queueTitle.first, '');
-        expect(await proxy.mediaItem.first, null);
-        expect(proxy.ratingStyle.hasValue, false);
-        expect(proxy.androidPlaybackInfo.hasValue, false);
-        expect(proxy.customState.hasValue, false);
-      });
-
-      test('method invocations $isolateLabel', () async {
-        Future<void> testMethod(String method, dynamic Function() function,
-            {dynamic expectedResult}) async {
-          final startCount = await proxy.count(method);
+    group('$isolateLabel method invocations $asciiSquare', () {
+      void testMethod(
+        String method,
+        dynamic Function() function, {
+        List<Object?> expectedArguments = const [],
+        dynamic expectedResult,
+      }) {
+        test('$method', () async {
```

Comment:
I made the `testMethod` rather create the `test`, because it was hard to debug and track which test exactly failed